File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function mergeTwoLists(list1: ListNode | null, list2: ListNode | null): ListNode | null {
    const dummy = new ListNode(0)
    let node = dummy
    while (list1 && list2) {
        if (list1.val < list2.val) {
            node.next = new ListNode(list1.val)
            list1 = list1.next
        } else {
            node.next = new ListNode(list2.val)
            list2 = list2.next
        }
        node = node.next
    }

    node.next = list1 ?? list2

    return dummy.next
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun mergeTwoLists(list1: ListNode?, list2: ListNode?): ListNode? {
        val dummy = ListNode(0);
        var node = dummy
        var node1 = list1;
        var node2 = list2;
        while (node1 != null && node2 != null) {
            if (node1.`val` < node2.`val`) {
                node.next = node1
                node1 = node1.next
            } else {
                node.next = node2
                node2 = node2.next
            }
            node = node.next!!
        }

        node.next = node1 ?: node2

        return dummy.next;
    }
}